Crash when player service is being stopped in some cases (NullPointerException)

### Checklist

- [X] I am able to reproduce the bug with the latest version given here: [CLICK THIS LINK](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.27.0

### Steps to reproduce the bug

1. Begin playing a video
2. Switch from WiFi to 5G or vice versa. Wait a moment

### Expected behavior

Video continues to play

### Actual behavior

Video stops playing and NewPipe opens up (even when playing in the background) with the guru meditation screen.

I'll need to close and open the app to recover. Also sometimes my position in the video will be lost, so even going to it through the History tab won't allow me to resume where I left off. That makes it even more annoying.

### Screenshots/Screen recordings

_No response_

### Logs

## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ US
* __Content Language:__ en-US
* __App Language:__ en_US
* __Service:__ none
* __Version:__ 0.27.0
* __OS:__ Linux Android 13 - 33
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: Unable to stop service org.schabi.newpipe.player.PlayerService@1ac624e: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.google.android.exoplayer2.ext.mediasession.MediaSessionConnector.setCustomActionProviders(com.google.android.exoplayer2.ext.mediasession.MediaSessionConnector$CustomActionProvider[])' on a null object reference
	at android.app.ActivityThread.handleStopService(ActivityThread.java:5285)
	at android.app.ActivityThread.-$$Nest$mhandleStopService(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2454)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:226)
	at android.os.Looper.loop(Looper.java:313)
	at android.app.ActivityThread.main(ActivityThread.java:8772)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.google.android.exoplayer2.ext.mediasession.MediaSessionConnector.setCustomActionProviders(com.google.android.exoplayer2.ext.mediasession.MediaSessionConnector$CustomActionProvider[])' on a null object reference
	at org.schabi.newpipe.player.mediasession.MediaSessionPlayerUi.updateMediaSessionActions(MediaSessionPlayerUi.java:215)
	at org.schabi.newpipe.player.mediasession.MediaSessionPlayerUi.onMetadataChanged(MediaSessionPlayerUi.java:282)
	at org.schabi.newpipe.player.Player.lambda$updateMetadataWith$22(Player.java:1801)
	at org.schabi.newpipe.player.Player.$r8$lambda$6LL1ynLVHUrHxxol0bvWL1Rn_TU(Player.java:0)
	at org.schabi.newpipe.player.Player$$ExternalSyntheticLambda47.accept(R8$$SyntheticClass:0)
	at j$.util.AbstractList$RandomAccessSpliterator.forEachRemaining(Unknown Source:19)
	at j$.util.stream.ReferencePipeline$Head.forEachOrdered(Unknown Source:10)
	at org.schabi.newpipe.player.ui.PlayerUiList.call(PlayerUiList.java:88)
	at org.schabi.newpipe.player.Player.updateMetadataWith(Player.java:1801)
	at org.schabi.newpipe.player.Player.lambda$onEvents$13(Player.java:1293)
	at org.schabi.newpipe.player.Player.$r8$lambda$5dalatdJw7zsnq99HOZvtSHTxZo(Player.java:0)
	at org.schabi.newpipe.player.Player$$ExternalSyntheticLambda37.accept(R8$$SyntheticClass:0)
	at j$.util.Optional.ifPresent(Unknown Source:4)
	at org.schabi.newpipe.player.Player.lambda$onEvents$14(Player.java:1287)
	at org.schabi.newpipe.player.Player.$r8$lambda$67JZJPUrqqIwv2qPVBiXIeF-1Rs(Player.java:0)
	at org.schabi.newpipe.player.Player$$ExternalSyntheticLambda9.accept(R8$$SyntheticClass:0)
	at j$.util.Optional.ifPresent(Unknown Source:4)
	at org.schabi.newpipe.player.Player.onEvents(Player.java:1265)
	at com.google.android.exoplayer2.ExoPlayerImpl.lambda$new$0(ExoPlayerImpl.java:266)
	at com.google.android.exoplayer2.ExoPlayerImpl.$r8$lambda$FaN7Odt_OJn5xWFlmhnR2OmiNMg(ExoPlayerImpl.java:0)
	at com.google.android.exoplayer2.ExoPlayerImpl$$ExternalSyntheticLambda1.invoke(R8$$SyntheticClass:0)
	at com.google.android.exoplayer2.util.ListenerSet$ListenerHolder.release(ListenerSet.java:320)
	at com.google.android.exoplayer2.util.ListenerSet.remove(ListenerSet.java:183)
	at com.google.android.exoplayer2.ExoPlayerImpl.removeListener(ExoPlayerImpl.java:1641)
	at org.schabi.newpipe.player.Player.destroyPlayer(Player.java:573)
	at org.schabi.newpipe.player.Player.destroy(Player.java:600)
	at org.schabi.newpipe.player.PlayerService.cleanup(PlayerService.java:150)
	at org.schabi.newpipe.player.PlayerService.onDestroy(PlayerService.java:145)
	at android.app.ActivityThread.handleStopService(ActivityThread.java:5265)
	... 9 more

```
</details>
<hr>

### Affected Android/Custom ROM version

_No response_

### Affected device model

Samsung Galaxy ZFold 4

### Additional information

Phone: SM-F936U1
One UI 5.0
Android 13
Kernel Version: 5.10.81-android12-9-25407272-abF936U1UEU1BVKB

This is a de-Googled phone, so it doesn't run Google Play Services, which has been put to deep sleep via adb.

Maybe you can add a null check here: https://github.com/TeamNewPipe/NewPipe/blob/879d7a24f05ab06510be249f2f3b5109f7c89dcc/app/src/main/java/org/schabi/newpipe/player/mediasession/MediaSessionPlayerUi.java#L215

```
private void updateMediaSessionActions() {
    if (sessionConnector == null) {
        // log...
        return;
    }

    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) {
        return;
    }

    // ...
}
```

Or maybe it's clear to you why https://github.com/TeamNewPipe/NewPipe/blob/879d7a24f05ab06510be249f2f3b5109f7c89dcc/app/src/main/java/org/schabi/newpipe/player/mediasession/MediaSessionPlayerUi.java#L280 can be called when init apparently wasn't called, or `sessionConnector` is somehow being unset?

This happens very often to me, basically whenever I leave the house while playing a video in the background, or entering.

I got the same error, but in a different situation. I was watching a video and wanted to play another one. I clicked on the other one. Then the message appeared. After clicking away the error message, I had to restart NewPipe.

Smartphone: Fairphone 4 (CalyxOS 5.6.3, Android 14)
NewPipe version 0.27.0

## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ US
* __Content Language:__ en-US
* __App Language:__ en_US_#u-fw-mon-mu-celsius
* __Service:__ none
* __Version:__ 0.27.0
* __OS:__ Linux Android 14 - 34
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: Unable to stop service org.schabi.newpipe.player.PlayerService@46f479d: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.google.android.exoplayer2.ext.mediasession.MediaSessionConnector.setCustomActionProviders(com.google.android.exoplayer2.ext.mediasession.MediaSessionConnector$CustomActionProvider[])' on a null object reference
	at android.app.ActivityThread.handleStopService(ActivityThread.java:5085)
	at android.app.ActivityThread.-$$Nest$mhandleStopService(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2398)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8501)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.google.android.exoplayer2.ext.mediasession.MediaSessionConnector.setCustomActionProviders(com.google.android.exoplayer2.ext.mediasession.MediaSessionConnector$CustomActionProvider[])' on a null object reference
	at org.schabi.newpipe.player.mediasession.MediaSessionPlayerUi.updateMediaSessionActions(MediaSessionPlayerUi.java:215)
	at org.schabi.newpipe.player.mediasession.MediaSessionPlayerUi.onMetadataChanged(MediaSessionPlayerUi.java:282)
	at org.schabi.newpipe.player.Player.lambda$updateMetadataWith$22(Player.java:1801)
	at org.schabi.newpipe.player.Player.$r8$lambda$6LL1ynLVHUrHxxol0bvWL1Rn_TU(Player.java:0)
	at org.schabi.newpipe.player.Player$$ExternalSyntheticLambda47.accept(R8$$SyntheticClass:0)
	at j$.util.AbstractList$RandomAccessSpliterator.forEachRemaining(Unknown Source:19)
	at j$.util.stream.ReferencePipeline$Head.forEachOrdered(Unknown Source:10)
	at org.schabi.newpipe.player.ui.PlayerUiList.call(PlayerUiList.java:88)
	at org.schabi.newpipe.player.Player.updateMetadataWith(Player.java:1801)
	at org.schabi.newpipe.player.Player.lambda$onEvents$13(Player.java:1293)
	at org.schabi.newpipe.player.Player.$r8$lambda$5dalatdJw7zsnq99HOZvtSHTxZo(Player.java:0)
	at org.schabi.newpipe.player.Player$$ExternalSyntheticLambda37.accept(R8$$SyntheticClass:0)
	at j$.util.Optional.ifPresent(Unknown Source:4)
	at org.schabi.newpipe.player.Player.lambda$onEvents$14(Player.java:1287)
	at org.schabi.newpipe.player.Player.$r8$lambda$67JZJPUrqqIwv2qPVBiXIeF-1Rs(Player.java:0)
	at org.schabi.newpipe.player.Player$$ExternalSyntheticLambda9.accept(R8$$SyntheticClass:0)
	at j$.util.Optional.ifPresent(Unknown Source:4)
	at org.schabi.newpipe.player.Player.onEvents(Player.java:1265)
	at com.google.android.exoplayer2.ExoPlayerImpl.lambda$new$0(ExoPlayerImpl.java:266)
	at com.google.android.exoplayer2.ExoPlayerImpl.$r8$lambda$FaN7Odt_OJn5xWFlmhnR2OmiNMg(ExoPlayerImpl.java:0)
	at com.google.android.exoplayer2.ExoPlayerImpl$$ExternalSyntheticLambda1.invoke(R8$$SyntheticClass:0)
	at com.google.android.exoplayer2.util.ListenerSet$ListenerHolder.release(ListenerSet.java:320)
	at com.google.android.exoplayer2.util.ListenerSet.remove(ListenerSet.java:183)
	at com.google.android.exoplayer2.ExoPlayerImpl.removeListener(ExoPlayerImpl.java:1641)
	at org.schabi.newpipe.player.Player.destroyPlayer(Player.java:573)
	at org.schabi.newpipe.player.Player.destroy(Player.java:600)
	at org.schabi.newpipe.player.PlayerService.cleanup(PlayerService.java:150)
	at org.schabi.newpipe.player.PlayerService.onDestroy(PlayerService.java:145)
	at android.app.ActivityThread.handleStopService(ActivityThread.java:5065)
	... 9 more

```
</details>
<hr>

--- a/app/src/main/java/org/schabi/newpipe/player/mediasession/MediaSessionPlayerUi.java
+++ b/app/src/main/java/org/schabi/newpipe/player/mediasession/MediaSessionPlayerUi.java
@@ -38,7 +38,9 @@ public class MediaSessionPlayerUi extends PlayerUi
         implements SharedPreferences.OnSharedPreferenceChangeListener {
     private static final String TAG = "MediaSessUi";
 
+    @Nullable
     private MediaSessionCompat mediaSession;
+    @Nullable
     private MediaSessionConnector sessionConnector;
 
     private final String ignoreHardwareMediaButtonsKey;
@@ -195,6 +197,11 @@ public class MediaSessionPlayerUi extends PlayerUi
             // cause any trouble, it also doesn't seem to do anything, so we don't do anything to
             // save battery. Check out NotificationUtil.updateActions() to see what happens on
             // older android versions.
+            return;
+        }
+
+        if (sessionConnector == null) {
+            // sessionConnector will be null after destroyPlayer is called
             return;
         }
 

